### PR TITLE
Issue40-integrate-pagination

### LINKED ISSUE #40: Integrate Pagination from azure-db into finesse-backend
Following the implementation of pagination in Azure DB to address significant latency issues with large result sets, we need to update the Finesse-Backend to properly use the enhanced Azure DB functionality. This involves adjusting our backend services to handle pagination parameters and ensure seamless integration with the frontend.

### Tasks
- [x] Update the Finesse-Backend to accept and pass pagination parameters (page number and page size) to Azure DB search queries.
- [x] Ensure the backend correctly handles the paginated response from Azure DB and formats it for frontend consumption.
- [x] Conduct thorough testing to verify that the backend properly interacts with the paginated Azure DB search, focusing on both functionality and improved performance.
- [x] Update the documentation

### Acceptance Criteria
- Finesse-Backend is fully compatible with the paginated search functionality of Azure DB, maintaining or improving current search performance.
- Backend modifications support pagination, allowing for efficient data retrieval and minimal latency, with search response times consistently below 1 second in wort case scenarios.
- The integration is documented, detailing how the backend manages paginated data and interfaces with both Azure DB and the frontend.

Ref: https://github.com/ai-cfia/azure-db/issues/16

### PR DESCRIPTION
**Pull request by k-allagbe**
Closes #40: integrate pagination

- [x] updated azure route and search method call
- [x] refactored flask config to be a dict instead of an object
- [x] updated envs in `.env.template`
- [x] updated tests
- [x] updated docs
- [x] point `azure-db` dependency to the main branch

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -23,49 +21,89 @@
 )
 DEFAULT_HIGHLIGHT_FIELDS = "content"
 DEFAULT_HIGHLIGHT_TAG = "strong"
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply app/config.py]: Done. Note that I moved the constants into a separate file. I also removed unused constants.

File: tests/test_azure_search.py [hunk 2]
```diff
@@ -1,23 +1,75 @@
...
+                "/search/azure",
+                json={"query": "azure query"},
+                query_string={"skip": 1, "top": 5},
+            )
             self.assertEqual(response.status_code, 200)
             self.assertEqual(response.json, {"some": "azure data"})
+            mock_search.assert_called_with(
+                "azure query",
+                self.config["AZURE_SEARCH_CLIENT"],
+                {"skip": 1, "top": 5, **self.config["AZURE_SEARCH_PARAMS"]},
```
Comment (ibrahim-kabir) [inline tests/test_azure_search.py:50]: I don't see a test with top value at 0. Top at 0 means unlimited pages ?

Comment (k-allagbe) [reply tests/test_azure_search.py:50]: I will add a test for this in `azure-db` in a next PR.

**Review by ibrahim-kabir — APPROVED** [review]
LGTM

**Pull request merged**